### PR TITLE
MDM Context Map

### LINKED ISSUE #108: Context map

### PR DESCRIPTION
**Pull request by vitlinda**
This PR introduces the context map of the mdm bounded contexts and its description.
Closes #108

### CODE REVIEW

File: docs/_docs/context-map.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  Since `ProductionPlannig` is a downstream core bounded context, and Anti-Corruption Layer is required.
+- `Stocking [D, CF] <- [U] Production`  
+  `Production` informs `Stocking` that a batch is ripening.
+  Since `Production` and `Stocking` are tightly coupled, the latter is Conformist.
+- `Production [D, ACL] <- [U] Restocking`  
+  `Production` informs `Restocking` when some raw materials are consumed.
+  `Production` is a downstream bounded context and needs an Anti-Corruption Layer since `Restocking` is going to be a generic bounded context.
+
+There is a *Shared Kernel* between every bounded context which contains the definitions for **product** and **cheese type**.
+This choice was taken as the two concepts above are crucial and every change in the shared kernel must affect every other bounded context.
```
Comment (giacomocavalieri) [inline docs/_docs/context-map.md]: I'd slightly rephrase this:
```suggestion
There is a *Shared Kernel* among the bounded contexts which contains the definitions for **product** and **cheese type**.
This choice was taken as the two aforementioned concepts are crucial for the cheese factory and a change in any of the definitions must be reflected in all bounded contexts handling these concepts.
In fact, adding a new kind of product involves a series of important domain changes that must be reflected in the code of different bounded contexts: the production, ordering, labelling and stocking processes would need a rehaul to take into account the new kind of product.

By sharing this information among different bounded context it is guaranteed that, whenever a change happens to any of these concepts, all the domains will maintain an up-to-date vision of these concepts.

```

**Review by giacomocavalieri — APPROVED** [review]

**Review by ndido98 — APPROVED** [review]

**Comment by nicolasfara** [comment]
Yesterday it turned out that `milk-planning` needs an interaction with `stocking` to find out how much product is available to make the milk reorder. No interaction between them appears in the diagram, can you fix it?

File: docs/_docs/context-map.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+- `Stocking [D, CF] <- [U] Production`  
+  `Production` informs `Stocking` that a batch is ripening.
+  Since `Production` and `Stocking` are tightly coupled, the latter is Conformist.
+- `Production [D, ACL] <- [U] Restocking`  
+  `Production` informs `Restocking` when some raw materials are consumed.
+  `Production` is a downstream bounded context and needs an Anti-Corruption Layer since `Restocking` is going to be a generic bounded context.
+- `MilkPlanning [D, ACL] <- [U] Stocking`  
+  `MilkPlanning` asks to `Stocking` the current stock.
+  `MilkPlanning` is a downstream core domain since `Stocking` provides a service to it, so the former has an Anti-Corruption Layer on its side.
+
```
Comment (nicolasfara) [inline docs/_docs/context-map.md:37]: Yesterday it turned out that `milk-planning` needs an interaction with `stocking` to find out how much product is available to make the milk reorder. No interaction between them appears in the diagram, can you fix it?

**Review by nicolasfara — CHANGES_REQUESTED** [review]

**Review by nicolasfara — CHANGES_REQUESTED** [review]
In the image there is no interaction between `stocking` and `milk-planning`

**Review by nicolasfara — APPROVED** [review]

**Pull request merged**